### PR TITLE
Add directions to map

### PR DESCRIPTION
**Pull request by scottpenneman**
The browser will obtain user location if allowed, and when a marker is clicked on the map, directions from the user location to the marker are displayed below the map.

I also updated the infoWindow with a link to Google Maps, which sends the user location and marker coordinates directly to Google Maps. I thought this would be useful as you can send directions to your phone from the Google Maps page. Also, if the link is opened on a cell phone, I believe it would open the Google Maps app if installed.

### CODE REVIEW

**Review by MulesKC — COMMENTED** [review]
Great work Scott! That map is too cool. All of the functionality worked for me. 

I did notice that you were missing the latest commit from master in this branch. I think this pull request will still work, but wanted to make sure you were going this route on purpose.

File: src/main/resources/static/styles.css [hunk 1]
```diff
@@ -32,6 +32,6 @@
 
 #map {
     height: 80vh;
-    width: 80vw;
+    width: 75vw;
```
Comment (KBeas21) [inline src/main/resources/static/styles.css]: why did this width change?

Comment (scottpenneman) [reply src/main/resources/static/styles.css]: Ah, I was messing with placement of the directions, and did not mean to leave it at 75. Fixed!

**Comment by scottpenneman** [comment]
> Great work Scott! That map is too cool. All of the functionality worked for me.
> 
> I did notice that you were missing the latest commit from master in this branch. I think this pull request will still work, but wanted to make sure you were going this route on purpose.

**Review by KBeas21 — APPROVED** [review]

**Review by jamiander — APPROVED** [review]

**Pull request merged**